sphinx/builders/html/__init__.py

```diff
@@ -425,7 +425,7 @@ def render_partial(self, node: Node | None) -> dict[str, str]:
         if node is None:
             return {'fragment': ''}
 
-        doc = new_document('<partial node>')
+        doc = new_partial_document()
```

Comment:
Perhaps just setting an attribute?

```suggestion
        doc = new_document('<partial node>')
        doc['is_partial'] = True
```

and then ``doc.get('is_partial', False)`` to test.

Reply:
I wanted to keep the old behaviour in case there was someone somewhere who relied on that. Also, it's to remind us that we have possible partial documents somewhere and that we always need to be aware of. 

On the other hand, using an attribute *may* lead false positive where a user would define this specific attribute. When using the source + special name we may avoid these cases (is_partial may be a "simple" attribute name that may be used in some projects).

Reply:
I'm averse to introducing new infrastructure around this if it isn't something we want to keep long-term. So I would prefer the attribute approach (keeping the document name the same), we could use ``_is_partial`` or some mangled name if we need to.

A

Reply:
> we could use _is_partial or some mangled name if we need to.

That's fine by me then! Actually I don't know whether we'll keep it long-term or not since I don't know if we will be able to fix this issue on partial documents. 

I'll do the update next week (I'me quite busy these days).